lightcurve_fitting/lightcurve.py

```diff
@@ -534,43 +534,53 @@ def mag2flux(mag, dmag=np.nan, zp=0., nondet=None, nondetSigmas=3.):
         flux[nondet] = 0
     return flux, dflux
 
-
-def binflux(time, flux, dflux, delta=0.2):
+def binflux(time, flux, dflux, delta=0.2, include_zero=True):
     """
     Bin a light curve by averaging points within ``delta`` of each other in time
-
     Parameters
     ----------
     time, flux, dflux : array-like
         Arrays of times, fluxes, and uncertainties comprising the observed light curve
     delta : float, optional
         Bin size, in the same units as ``time``. Default: 0.2
-
+    include_zero : bool, optional
+        Include data points with no error bar
     Returns
     -------
     time, flux, dflux : array-like
         Binned arrays of times, fluxes, and uncertainties
     """
-    bin_time = []
-    bin_flux = []
+    bin_time  = []
+    bin_flux  = []
     bin_dflux = []
     while len(flux) > 0:
         grp = np.array(abs(time - time[0]) <= delta)
-        time_grp = time[grp]
-        flux_grp = flux[grp]
+        time_grp  = time[grp]
+        flux_grp  = flux[grp]
         dflux_grp = dflux[grp]
-        if any(dflux_grp == 0) or any(dflux_grp == 9999) or any(np.isnan(dflux_grp)) or (
-                np.ma.is_masked(dflux_grp) and any(dflux_grp.mask)):
+
+        # Indices with no error bar
+        zeros = ((dflux_grp == 0) | (dflux_grp == 999) | (dflux_grp == 9999) | (dflux_grp == -1) | np.isnan(dflux_grp) | 
+                (np.ma.is_masked(dflux_grp) and any(dflux_grp.mask)))
```

Comment:
I don't think this will do exactly what you want. We want to exclude elements where `dflux_grp.mask = True`, but the problem is `dflux_grp` may not have a mask, which is what the `np.ma.is_masked` is checking for. It might not be possible to do this in a single line anymore. Maybe separate the last part into
```
if np.ma.is_masked(flux_grp):
    zeros |= dflux_grp.mask
```

Reply:
Hopefully this is fixed now